### PR TITLE
Image templates rebuild

### PR DESCRIPTION
**Pull request by Ana06**
Disable publish for image template branches. Set repository rebuild attribute to local when branching image template repositories. A separated method was created for the image_template related code.

Comment the `global_write_through` variable set to `true` line as it is in the other spec files.

### CODE REVIEW

**Comment by Ana06** [comment]
[(5) Branching image templates (api)](https://trello.com/c/qZRROUHe/96-5-branching-image-templates-api)

**Comment by Ana06** [comment]
[(5) Branching image templates (api)](https://trello.com/c/qZRROUHe/96-5-branching-image-templates-api)

**Comment by adrianschroeter** [comment]
We should just add a branch option to copy rebuild startegies. We may use it by default when coping a template, but we should enforce a concrete strategy in the code, because we may want to have a different one for some templates.

So, just copying the strategy from the template project should be enough.

We should also change "image_template" to "template" since I don't see any image specific stuff here. The same would also work with spec/dsc/PKGBUILD/... no?

**Comment by hennevogel** [comment]
@adrianschroeter and which sense would this make for package descriptions?

**Comment by adrianschroeter** [comment]
Why not have templates for perl, python, rubygem, ... ?
Maybe something where you just need to add a module name and the package would do the rest.

I would not exclude it at least, if the implementation has no limitations to images.

File: src/api/spec/models/branch_package_spec.rb [hunk 1]
```diff
@@ -1,6 +1,8 @@
 require 'rails_helper'
-
-CONFIG['global_write_through'] = true
+# WARNING: If you change tests make sure you uncomment this line
+# and start a test backend. Some of the actions
+# require real backend answers for projects/packages.
+# CONFIG['global_write_through'] = true
```
Comment (bgeuken) [inline src/api/spec/models/branch_package_spec.rb]: Please squash this commit into c07031933180908dd4d09bc1f6c8f63ccfbda1db

**Comment by hennevogel** [comment]
@adrianschroeter but package descriptions have nothing to do with this feature. All the things we do differently to a normal package branch are for image templates only and make no sense for package descriptions. We can have package templates later and do other things differently to them, not now.

File: src/api/spec/models/branch_package_spec.rb (hunk 1 above)
Comment (bgeuken) [inline src/api/spec/models/branch_package_spec.rb]: @Ana06 ping:-)

**Comment by bgeuken** [comment]
> We should just add a branch option to copy rebuild strategies. We may use it by default when coping a 
> template, but we should enforce a concrete strategy in the code, because we may want to have a
> different one for some templates.

I am not sure if I can follow. What do you mean with "add a branch strategy"?
And which strategy is it you would enforce?
@adrianschroeter

**Comment by adrianschroeter** [comment]
After some discussion with Björn:

We don't need the logic here, it is only important that it is possible to branch in a way that we get the auto cleanup and the right rebuild strategy applied. Hardcode the policy here would create problems later.

So, we can say that the needed functionality exists already and we can just skip this commit.
Sorry.

(I have extended the UI branch with another criteria which we need to discuss in next meeting).

**Comment by bgeuken** [comment]
I guess it's okay to close this one then.

**Pull request closed without merging**